### PR TITLE
feat: improve audit logs middleware

### PR DESCRIPTION
**Pull request by wI2L**
Signed-off-by: William Poussier <william.poussier@ovhcloud.com>

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Refactor of the error logs middleware.

* **What is the current behavior?** (You can also link to an open issue here)

Only errors with a few fields are logged.

* **What is the new behavior (if this is a feature change)?**

This logs more metadata, such as the action name, user, ...

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No.

### CODE REVIEW

File: api/middleware.go [hunk 1]
```diff
@@ -2,25 +2,59 @@ package api
...
+	q, _ := url.QueryUnescape(c.Request.URL.RawQuery)
+
+	fields := logrus.Fields{
+		"status":          c.Writer.Status(),
+		"method":          c.Request.Method,
+		"path":            c.Request.URL.Path,
+		"query":           q,
+		"user_agent":      c.Request.UserAgent(),
+		"duration":        requestDuration.Seconds(),
+		"duration_ms":     requestDuration.Seconds() * 1e3,
```
Comment (rbeuque74) [inline api/middleware.go:35]:
```suggestion
		"duration_ms":     requestDuration.Milliseconds(),
```

**Review by rbeuque74 — COMMENTED** [review]
Rest is OK.

- [x] Need to add the metadata actions in handlers.
- [ ] Resolution debugging logs could also use some fields and Gelf format (and adding a field `log_type` to either `access_log` for the log of handlers, and `engine` for debugging logs to be able to differentiate them)
- [x] Need to be able to configure the logging format (GELF for example)

**Comment by ovh-cds** [comment]
CDS Report test#1050.0 ✘  
* tests and lint
  * golangci-lint ✘  
  * unit-test and integration-tests ✔

**Comment by ovh-cds** [comment]
CDS Report test#1051.0 ✘  
* tests and lint
  * golangci-lint ✘  
  * unit-test and integration-tests ✔

**Comment by ovh-cds** [comment]
CDS Report test#1052.0 ✘  
* tests and lint
  * golangci-lint ✘  
  * unit-test and integration-tests ✔

**Comment by ovh-cds** [comment]
CDS Report package#1054.0 ✘  
* Stage 1
  * Docker build ✘

**Comment by ovh-cds** [comment]
CDS Report test#1055.0 ✘  
* tests and lint
  * golangci-lint ✔ 
  * unit-test and integration-tests ✘

**Comment by ovh-cds** [comment]
CDS Report test#1056.0 ✘  
* tests and lint
  * golangci-lint ✔ 
  * unit-test and integration-tests ✘

**Review by rbeuque74 — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report test#1057.0 ✘  
* tests and lint
  * golangci-lint ✔ 
  * unit-test and integration-tests ✘

**Comment by ovh-cds** [comment]
CDS Report test#1058.0 ✘  
* tests and lint
  * golangci-lint ✔ 
  * unit-test and integration-tests ✘

**Review by rbeuque74 — DISMISSED** [review]

File: api/handler/comment.go [hunk 2]
```diff
@@ -186,8 +209,22 @@ func UpdateComment(c *gin.Context, in *updateCommentIn) (*task.Comment, error) {
 
 	reqUsername := auth.GetIdentity(c)
 
-	if auth.IsAdmin(c) != nil && reqUsername != comment.Username {
-		return nil, errors.Forbiddenf("Can't update comment")
+	admin := auth.IsAdmin(c) == nil
+	commentAuthor := reqUsername == comment.Username
```
Comment (wI2L) [inline api/handler/comment.go:213]: `commentAuthor` could be easier to read as `isCommentAuthor`.

File: engine/engine.go [hunk 3]
```diff
@@ -211,6 +211,8 @@ func (e Engine) launchResolution(publicID string, async bool, sm *semaphore.Weig
 		return nil, nil
 	}
 
+	debugLogger = debugLogger.WithFields(logrus.Fields{"template_name": t.TemplateName, "task_id": t.PublicID})
```
Comment (wI2L) [inline engine/engine.go:214]: Should use the constants added in `pkg/metadata` package as field names in this file.

**Review by wI2L — COMMENTED** [review]
LGTM, small fixes remaining.

**Review by rbeuque74 — APPROVED** [review]

**Pull request merged**